nucleus/dataset.py

```diff
@@ -1677,3 +1677,23 @@ def update_item_metadata(self, mapping: Dict[str, dict]):
             self.id, self._client, mapping, ExportMetadataType.DATASET_ITEMS
         )
         return mm.update()
+
+    def list_query_items(self, query: str) -> Iterable[DatasetItem]:
```

Comment:
Think this is a bit cleaner: 
```suggestion
    def query_items(self, query: str) -> Iterable[DatasetItem]:
```

Reply:
I jotted down some thoughts on unified pagination [in this doc](https://docs.google.com/document/d/1jJRIkRG2RBV1Am10vB3hRbTUAJXcGV5czLx8J3CtiAU/edit#)!

Relevant point here being, imo we should have a more consistent naming scheme for these "paginated list export" type endpoints. e.g.
- `list_items`
- `list_item_annotations`
- `list_query_items` <- not sure though, agree it is a bit awkward

But yes given that there isn't much of a naming convention at the moment, `query_items` sftm 😅